hudi-timeline-service/src/main/java/org/apache/hudi/timeline/service/handlers/MarkerHandler.java

```diff
@@ -0,0 +1,482 @@
+                  bufferedReader = new BufferedReader(new InputStreamReader(fsDataInputStream, StandardCharsets.UTF_8));
+                  markers = bufferedReader.lines().collect(Collectors.toSet());
+                  bufferedReader.close();
+                  fsDataInputStream.close();
+                } catch (IOException e) {
+                  throw new HoodieIOException("Failed to read MARKERS file " + markerDirPath, e);
+                } finally {
+                  closeQuietly(bufferedReader);
+                  closeQuietly(fsDataInputStream);
+                }
+                return new ImmutablePair<>(markersFilePathStr, markers);
+              }, actualParallelism);
+
+          Set<String> allMarkers = new HashSet<>();
+          for (String markersFilePathStr: fileMarkersSetMap.keySet()) {
+            Set<String> fileMarkers = fileMarkersSetMap.get(markersFilePathStr);
+            if (!fileMarkers.isEmpty()) {
+              Map<Integer, StringBuilder> singleDirMarkersMap =
+                  fileMarkersMap.computeIfAbsent(markerDir, k -> new HashMap<>());
+              int index = getMarkerFileIndex(markersFilePathStr);
+
+              if (index >= 0) {
+                singleDirMarkersMap.put(index, new StringBuilder(StringUtils.join(",", fileMarkers)));
+                allMarkers.addAll(fileMarkers);
+              }
+            }
+          }
+          allMarkersMap.put(markerDir, allMarkers);
+        }
+      }
+    } catch (IOException ioe) {
+      throw new HoodieIOException(ioe.getMessage(), ioe);
+    }
+  }
+
+  /**
+   * Closes {@code Closeable} quietly.
+   *
+   * @param closeable {@code Closeable} to close
+   */
+  private void closeQuietly(Closeable closeable) {
+    if (closeable == null) {
+      return;
+    }
+    try {
+      closeable.close();
+    } catch (IOException e) {
+      LOG.warn("IOException during close", e);
+    }
+  }
+
+  /**
+   * Future for async marker creation request.
+   */
+  private class CreateMarkerCompletableFuture extends CompletableFuture<String> {
+    private final Context context;
+    private final String markerDirPath;
+    private final String markerName;
+    private boolean result;
+    private final long startTimeMs;
+
+    public CreateMarkerCompletableFuture(Context context, String markerDirPath, String markerName) {
+      super();
+      this.startTimeMs = System.currentTimeMillis();
+      this.context = context;
+      this.markerDirPath = markerDirPath;
+      this.markerName = markerName;
+      this.result = false;
+    }
+
+    public Context getContext() {
+      return context;
+    }
+
+    public String getMarkerDirPath() {
+      return markerDirPath;
+    }
+
+    public String getMarkerName() {
+      return markerName;
+    }
+
+    public boolean getResult() {
+      return result;
+    }
+
+    public void setResult(boolean result) {
+      LOG.info("Request queued for " + (System.currentTimeMillis() - startTimeMs) + " ms");
```

Comment:
Can we switch this to debug.

Reply:
Yes.  I revisited all logs and changed some of them to debug level if appropiate.